src/native/configuration/globalconfig.rs

```diff
@@ -136,15 +133,15 @@ impl GlobalConfig {
 
     /// Initialize from file on disk
     pub async fn from_disk() -> Result<Self, NativeError> {
-        if let Ok(file) = get_read(&config_path())
-            && let Ok(config) = serde_json::from_reader(file)
-        {
-            Ok(config)
-        } else {
-            let config = Self::create().await?;
-            config.to_disk()?;
-            Ok(config)
+        if let Ok(file) = get_read(&config_path()) {
+            if let Ok(config) = serde_json::from_reader(file) {
+                return Ok(config);
+            }
         }
+
+        let config = Self::create().await?;
+        config.to_disk()?;
+        Ok(config)
```

Comment:
These hidden failure modes are impossible to reason about without diving into the code. This is one of those patterns that needs to go